### PR TITLE
[11.0] mig base geoengine

### PR DESCRIPTION
**Pull request by asaunier**
Replaces initial PR at https://github.com/OCA/geospatial/pull/176
Requires changes in module base_geoengine_demo as in https://github.com/OCA/geospatial/pull/177

* [x] replay commits with git am
* [ ]  squash commits => at least the new commits are squashed
* [ ]  migrate base_geoengine
  * [x] py2 to py3
  * [x] make the geoengine view work
  * [x] make the main widget work (geo_edit_map)
  * [x] make the other geo widgets work
  * [ ] use inheritance of base model to create a mixin
* [ ] update OL version

Know issues:

* Multi selection filter is not displayed in the search view (need to find how to create it)
* A widget map in a `<notebook><page>` in a modal view won't be displayed, it is shown only if you switch to an other tab and come back on the first tab. An issue with event trigger is the cause.
* Minified version of openlayers is never used, it should when not in debug mode.
* WMTS has issue with reprojections.
* Currently updated to last version of OL3, OL4 and OL5 have some breaking changes we didn't dived into.
* Some travis issues (psycopg, js lint).

### CODE REVIEW

File: base_geoengine/geo_model.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+from odoo.exceptions import except_orm, MissingError
+from . import geo_operators
+from . import fields as geo_fields
+
+
+DEFAULT_EXTENT = ('-123164.85222423, 5574694.9538936, '
+                  '1578017.6490538, 6186191.1800898')
+
+
+class GeoModel(models.BaseModel):
```
Comment (lmignon) [inline base_geoengine/geo_model.py:14]: Do we still need this specialized model? Why not inheriting from `base`?

Comment (yvaucher) [reply base_geoengine/geo_model.py:14]: I think the same it was not a priority but I agree we can improve this. I'll add it in the todo list.

(a mixin is definitely the way to go)

**Comment by nikul-serpentcs** [comment]
@asaunier If it's WIP then change title like `[11.0] [WIP] base geoengine `

**Review by yvaucher — CHANGES_REQUESTED** [review]
There is a cursor error to fix in travis tests:

```
  File "/home/travis/OCB-11.0/odoo/sql_db.py", line 154, in wrapper
    raise psycopg2.OperationalError(msg)
psycopg2.OperationalError: Unable to use a closed cursor.
```

**Comment by asaunier** [comment]
@yvaucher Hopefully 56a359b should fix the closed cursor error reported by Travis.

**Comment by asaunier** [comment]
Linting errors have been fixed but Travis still reports an error with sphinxodoo:
> File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sphinxodoo/ext/autodoc.py", line 43, in load_modules
    import openerp
ImportError: No module named 'openerp'

It seems Travis is stuck with an old version of sphinxodoo (perhaps https://github.com/OCA/odoo-sphinx-autodoc/blob/8a3c37034349a2440f0b96fc6cc50ca877233d50/sphinxodoo/ext/autodoc.py#L43).

@yvaucher How could we make it use the last version that should handle both odoo/openerp?
See https://github.com/OCA/odoo-sphinx-autodoc/blob/master/sphinxodoo/ext/autodoc.py#L48-L52

By the way, could someone remove the "work in progress" label? @max3903 ?

**Review by yvaucher — APPROVED** [review]
LGTM for a first working version

Tests are green only the sphinx doc generation which is broken since ages makes the CI status red.

**Comment by asaunier** [comment]
@yvaucher Thanks for reviewing! I have squashed the last 2 fixup commits into 8fd3b3c

Please note this [additional PR](https://github.com/asaunier/geospatial/pull/1) by @smangukiya on top of the current PR. Perhaps to be added as a usual PR when this one is merged?

File: base_geoengine/doc/build/_static/default.css [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+.viewcode-back {
+    font-family: sans-serif;
+}
+
+div.viewcode-block:target {
+    background-color: #f4debf;
+    border-top: 1px solid #ac9;
+    border-bottom: 1px solid #ac9;
+}
```
Comment (nikul-serpentcs) [inline base_geoengine/doc/build/_static/default.css:256]:
``` suggestion
}

```

File: base_geoengine/doc/build/genindex.html [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+             >modules</a> |</li>
+        <li><a href="index.html">OpenERP GeoEngine 0.3 documentation</a> &raquo;</li> 
+      </ul>
+    </div>
+    <div class="footer">
+        &copy; Copyright 2012, Camptocamp, Nicolas Bessi.
+      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.2.
+    </div>
+  </body>
+</html>
```
Comment (nikul-serpentcs) [inline base_geoengine/doc/build/genindex.html:337]:
```suggestion
</html>

```

File: base_geoengine/security/data.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        <field name="name">Geoengine User</field>
+    </record>
+
+    <record id="group_geoengine_admin" model="res.groups">
+        <field name="name">Geoengine Admin</field>
+        <field name="implied_ids" eval="[(4, ref('group_geoengine_user')),]"/>
+        <field name="users" eval="[(4, ref('base.user_root'))]"/>
+    </record>
+
+</odoo>
```
Comment (nikul-serpentcs) [inline base_geoengine/security/data.xml:14]:
```suggestion
</odoo>

```

File: base_geoengine/doc/build/_static/default.css (hunk 2 above)
Comment (yvaucher) [reply base_geoengine/doc/build/_static/default.css:256]: Please don't use the suggested changes. It is missing EOL we don't want an extra line.

Comment (nikul-serpentcs) [reply base_geoengine/doc/build/_static/default.css:256]: > Please don't use the suggested changes. It is missing EOL we don't want an extra line.

@yvaucher Here If you did not put a new blank line then Git Error like this,

test2222222222


So better to put EOL

**Comment by max3903** [comment]
@asaunier Is it ready to be reviewed and merged or is it still a work in progress to complete all the other checkboxes (like update OL version)?

**Comment by asaunier** [comment]
@max3903 It's ready to be reviewed, there is currently no work in progress about this PR. Not sure the remaining checkboxes will be addressed in this PR.

**Review by max3903 — APPROVED** [review]

**Comment by max3903** [comment]
@nikul-serpentcs I took care of your feedbacks in #183.

**Review by lmignon — APPROVED** [review]

**Pull request merged**